src/OpenLoco/Audio/Audio.cpp

```diff
@@ -1122,79 +1122,55 @@ namespace OpenLoco::Audio
 
             // Load info on the song to play.
             const auto& mi = kMusicInfo[_currentSong];
-            auto buffer = loadMusicSample(mi.pathId);
-            if (_channels[enumValue(ChannelId::bgm)].load(*buffer))
-            {
-                _channels[enumValue(ChannelId::bgm)].setVolume(Config::get().volume);
-                if (!_channels[enumValue(ChannelId::bgm)].play(false))
-                {
-                    cfg.musicPlaying = 0;
-                }
-            }
-            else
-            {
-                cfg.musicPlaying = 0;
-            }
+            cfg.musicPlaying = playMusic(mi.pathId, cfg.volume, false);
 
             WindowManager::invalidate(WindowType::options);
         }
     }
 
-    // 0x0048AAD2
-    void resetMusic()
+    // 0x0048AC66
+    // previously called void playTitleScreenMusic()
+    bool playMusic(PathId sample, int32_t volume, bool loop)
     {
-        stopBackgroundMusic();
-        _currentSong = kNoSong;
-        _lastSong = kNoSong;
-    }
+        static PathId currentTrackPathId;
 
-    // 0x0048AAE8
-    void stopBackgroundMusic()
-    {
-        if (_audioInitialised && _channels[enumValue(ChannelId::bgm)].isPlaying())
+        auto* channel = getChannel(ChannelId::music);
+        if (!_audioInitialised || _audioIsPaused || !_audioIsEnabled || channel == nullptr || currentTrackPathId == sample)
```

Comment:
are you sure that last condition `currentTrackPathId == sample` should return false. I would think that should return `true` but do no work.

Reply:
good point, I'll make it return true